modules/misc/nix.nix

```diff
@@ -230,7 +230,9 @@ in {
         };
       };
 
-      "nix/nix.conf" = mkIf (cfg.settings != { }) { source = nixConf; };
+      "nix/nix.conf" = mkIf (cfg.settings != { } || cfg.extraOptions != "") {
```

Comment:
The change looks good, but the commit log might be more detailed about the use case it is adding support for.